web/src/dropdown_widget.ts

```diff
@@ -9,79 +10,125 @@ import render_inline_decorated_stream_name from "../templates/inline_decorated_s
-        default_id = null,
-        unique_id_type = null,
-        // Text to show if the current value is not in `get_options()`.
-        text_if_current_value_not_in_options = null,
-        hide_search_box = false,
-        // Disable the widget for spectators.
-        disable_for_spectators = false,
-    }) {
-        this.widget_name = widget_name;
-        this.widget_id = `#${CSS.escape(widget_name)}_widget`;
+    widget_name: string;
+    widget_id: string;
+    widget_wrapper_id: string;
+    widget_value_selector: string;
+    get_options: () => Option[];
+    item_click_callback: (
+        event: JQuery.ClickEvent,
+        instance: tippy.Instance,
+        widget: DropdownWidget,
+    ) => void;
+    focus_target_on_hidden: boolean;
+    on_show_callback: (instance: tippy.Instance) => void;
+    on_mount_callback: (instance: tippy.Instance) => void;
+    on_hidden_callback: (instance: tippy.Instance) => void;
+    on_exit_with_escape_callback: () => void;
+    render_selected_option: () => void;
+    tippy_props: Partial<tippy.Props>;
+    list_widget: ListWidgetType<Option, Option> | undefined;
+    instance: tippy.Instance | undefined;
+    default_id: string | number;
+    current_value: string | number;
+    unique_id_type: DataTypes;
+    $events_container: JQuery;
+    text_if_current_value_not_in_options: string;
+    hide_search_box: boolean;
+    disable_for_spectators: boolean;
+
+    constructor(options: DropdownWidgetOptions) {
+        this.widget_name = options.widget_name;
+        this.widget_id = `#${CSS.escape(this.widget_name)}_widget`;
         // A widget wrapper may not exist based on the UI requirement.
         this.widget_wrapper_id = `${this.widget_id}_wrapper`;
         this.widget_value_selector = `${this.widget_id} .dropdown_widget_value`;
-        this.get_options = get_options;
-        this.item_click_callback = item_click_callback;
-        this.focus_target_on_hidden = focus_target_on_hidden;
-        this.on_show_callback = on_show_callback;
-        this.on_mount_callback = on_mount_callback;
-        this.on_hidden_callback = on_hidden_callback;
-        this.on_exit_with_escape_callback = on_exit_with_escape_callback;
-        this.render_selected_option = render_selected_option;
-        this.tippy_props = tippy_props;
-        this.list_widget = null;
-        this.instance = null;
-        this.default_id = default_id;
-        this.current_value = default_id;
-        this.unique_id_type = unique_id_type;
-        this.$events_container = $events_container;
-        this.text_if_current_value_not_in_options = text_if_current_value_not_in_options;
-        this.hide_search_box = hide_search_box;
-        this.disable_for_spectators = disable_for_spectators;
+        this.get_options = options.get_options;
+        this.item_click_callback = options.item_click_callback;
+        this.focus_target_on_hidden = options.focus_target_on_hidden ?? false;
```

Comment:
You’ve changed the default from `true` to `false`.

Reply:
Fixed.